emitter.py

```diff
@@ -29,6 +29,11 @@
 control_char_re = re.compile('[%s]' % re.escape(control_chars))
 
 
+# Only enforced for the metrics API on our end, for now
+MAX_COMPRESSED_SIZE = 2 << 20  # 2MB, the backend should accept up to 3MB but let's be conservative here
+MAX_SPLIT_DEPTH = 3  # maximum depth of recursive calls to payload splitting function
```

Comment:
How did you arrive at this number? It's cool if it's arbitrary, clearly from the test it can split a giant object.